src/addIndex.js

```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```

Comment:
I don't want to change a signature I don't understand. Let's leave this for a separate pull request. :)